core/src/main/scala/org/scalafmt/util/TreeOps.scala

```diff
@@ -486,4 +486,12 @@ object TreeOps {
   // procedure syntax has decltpe: Some("")
   def isProcedureSyntax(defn: Defn.Def): Boolean =
     defn.decltpe.exists(_.tokens.isEmpty)
+
+  def isMultiline(tree: Tree): Boolean = tree.tokens.exists(_.is[LF])
```

Comment:
I am afraid that this approach may cause non-idempotent formatting in some cases. Example

```scala
// maxColumn                  |
def foo = 2
def looongLine = { baar(1111, 2222, 33333))) }

// format once
def foo = 2
def looongLine = {
  baar(1111, 2222, 33333)))
}

// format second time
def foo = 2

def looongLine = {
  baar(1111, 2222, 33333)))
}
```

I don't think it's possible to determine inside Router.scala if a tree is multiline or not. What we can do instead is to handle this inside FormatWriter, which runs after the line wrapping algorithm has chosen where to put newlines and spaces.